### ISSUE TITLE
Any comment about upcoming Git Rev News edition 118

### ISSUE THREAD

**Issue by chriscool** [git/git.github.io#741]
A currently mostly empty draft is there:

https://github.com/git/git.github.io/blob/master/rev_news/drafts/edition-118.md

Feel free to comment in this issue, suggest topics, suggest persons to interview, or use the edit button (that looks like a pen) to edit and create a pull request with the changes you would like.

Let's try to publish this edition around the end of December 2024!

Thanks!

cc @jnareb @mjaix @sivaraam @gitster @stepnem

**Comment by chriscool** [comment]
Two Outreachy interns have been selected and should start working on their internship on December 9th. They have sent emails to the mailing list:

- https://lore.kernel.org/git/CAPSxiM_V1X7HtAAX+dJLH7c-ScVCnLxvazgJL=BGeC+kuEAHWQ@mail.gmail.com/
- https://lore.kernel.org/git/CAGedMtc9aKKemVZCYnzPH9rgNfMHfvukwLW8KMof1ZD+gj5MRw@mail.gmail.com/

**Comment by chriscool** [comment]
@jnareb maybe we could add a link to [this LWN article about adding Rust to Git](https://lwn.net/Articles/998115/ )?

**Comment by jnareb** [comment]
> @jnareb maybe we could add a link to [this LWN article about adding Rust to Git](https://lwn.net/Articles/998115/)?

Yes, I agree that would be a good idea (perhaps with reference to Edition 82?).

**Comment by chriscool** [comment]
@jnareb maybe we could add a link to [this short one about upcoming Git 2.48](https://www.phoronix.com/news/Git-2.48-rc0) too?

**Comment by jnareb** [comment]
> @jnareb maybe we could add a link to [this short one about upcoming Git 2.48](https://www.phoronix.com/news/Git-2.48-rc0) too?

That's a good idea. Thanks @chriscool for the suggestion.

By the way, do you have any reference to discussion about choosing "Meson to be the primary build tool for Git in the future"?

**Comment by chriscool** [comment]
@jnareb I think most of the discussion about Meson happened in that thread:

https://lore.kernel.org/git/20241119-pks-meson-v8-0-809bf7f042f3@pks.im/#r

Do you mean that I should cover this in an article?

**Comment by jnareb** [comment]
> @jnareb I think most of the discussion about Meson happened in that thread:
> 
> https://lore.kernel.org/git/cover.1727881164.git.ps@pks.im/

Thanks, though that thread looks like quite far in the process, being v8 of _"Modernize the build system"_ series...

I have just now remembered that the idea of Git switching to a modern build system was discussed on Git Contributor's Summit 2024: https://lore.kernel.org/git/Zu2E3vIcTzywWOx3@nand.local/

> * _**Patrick**: three different build system; should get rid of at least one of them_
>    * _Should delete autoconf because it's not really maintained_
>    * _Think about a proper build system_
>    * _Obvious choice: cmake_
[...]
> * _**Patrick**: [Makefiles are] Non-standard_
>    * _Meson is nicer than cmake as an alternative_

I think I was one of people proposing autoconf as an addition to configuring Git via Makefile variables very early in the life of Git as a project - it made it much easier to create RPM spec file when compilation follows traditional 'configure && make && make install' steps.

But the Linux ecosystem have much changed since then.

For example Meson is 8 years younger (2013) than Git (2005). CMake is older than Git (2000), but not much.

I guess Build2 is too new for the consideration...

> Do you mean that I should cover this in an article?

If you don't have a better idea for a topic, and have time for some research - it would be IMVHO interesting topic.

**Comment by jnareb** [comment]
I see that the following promise from Git Contributor's Summit 2024 (https://lore.kernel.org/git/Zu2E3vIcTzywWOx3@nand.local/)

> * _**Jonathan**: Should we do a comparison of build systems in terms of what_
>   _we need from them on the list? Similar to 'Documentation/technical/unit-tests.txt'_
>    * _**Patrick**: I can write such a thing._

There is now [Documentation/technical/build-systems.txt](https://github.com/git/git/blob/master/Documentation/technical/build-systems.txt), written by Patrick Steinhardt (@pks-t).

**Comment by chriscool** [comment]
Actually I already started writing the blob post that GitLab will publish about the upcoming v2.48.0 and we might talk about the Meson build system if it is part of that release (otherwise it will hopefully be part of the next one). So I have started researching the subject a bit, and yeah around 2006 it looks like you [introduced](https://github.com/git/git/commit/556677144b55aad8457851a9019e86c3676bd422) and worked on the configure.ac file to add autoconf build support to Git.

I will see if I have time to work on an article for Git Rev News about it...

**Comment by sivaraam** [comment]
ve nudged Karthik to see if we could get an interview from him. Otherwise, I suppose we'll  have no interview for this edition.

**Comment by chriscool** [comment]
Thanks @sivaraam ! No worries if we have no interview for this edition.

**Comment by chriscool** [comment]
Draft email sent: https://lore.kernel.org/git/CAP8UFD3y_m1D4pF1rCZgmW+C_6JYrxjuyUnGAhKBgbek+zBdRQ@mail.gmail.com/

Sorry it took me a long time to write an article about the Meson build system.

**Comment by chriscool** [comment]
As I sent the draft email only a few minutes ago, I think it's fine if we publish this on Thursday January 2nd 2025 as I say in that email.

**Comment by jnareb** [comment]
My links have landed in 2b88cb1 .

**Comment by jnareb** [comment]
>
> Sorry it took me a long time to write an article about the new Meson build system and related discussions.

No worry @chriscool . Thanks for a great and well-researched article about the planned use of Meson build system for Git!

I have made a few minor additions to it in c789c28 .

**Comment by chriscool** [comment]
@jnareb thanks for the great links and the nice improvements to the Meson build system article!

**Comment by chriscool** [comment]
I added a small article about our Outreachy interns in 3cf8bf0a600759653e1c2ed51b90e075933c6f6d. In it we say that we are still looking for funding, but I just learned that GitHub will fund the internships. I will change the article to thank GitHub.

**Comment by chriscool** [comment]
Small article about our Outreachy interns updated in bdfe3e4c61b2961c27c73b1b3b7729c526750e28.

**Comment by mjaix** [comment]
Supplied some fixes and rephrasings in 6154855 [1] , please scrutinize.
IMHO a great edition, both regarding the article and the links!

[1] 6154855
    https://github.com/git/git.github.io/commit/6154855bc2c93a4aa15d910488d4c5eb0fc701e1

**Comment by chriscool** [comment]
@mjaix thanks for the nice fixes and rephrasings!

**Comment by mjaix** [comment]
@jnareb [1] thanks for correcting my corrections - nobody is perfect :-
), and working in this team is just fun!

BTW, regarding AWK vs awk ... I once memorized the spelling of cURL
(which is still the Wikipedia one),
but recently learned that Daniel Stenberg seems to have switched to the
simpler curl spelling
[jag skulle igentligen fråga honom] and I think we you should use that
one now.

Some years ago, we had a spelling reform in Germany, and some of the
"great new" ways of spelling 
even had the potential to change the meaning but were fortunately
revised in a second stage ...

[1] @jnareb https://github.com/jnareb

**Comment by chriscool** [comment]
Edition pusblished and annouce email sent: https://lore.kernel.org/git/CAP8UFD1p4YZ0GB-6ffF5YwMM-_MSusHdKd4LxFP5O9dFJrwqiA@mail.gmail.com/
